R-0.0.5 (#170)
bob-crutchley/notes: topics/nginx/modules/introduction/README.md

# Introduction
<!--TOC_START-->
### Contents
- [Overview](#overview)
- [Installation](#installation)
	- [Linux](#linux)
		- [CentOS/RHEL](#centosrhel)
		- [Debian/Ubuntu](#debianubuntu)
	- [Windows](#windows)
	- [Docker](#docker)
- [Tasks](#tasks)
	- [Install NGINX](#install-nginx)

<!--TOC_END-->
## Overview
Nginx is a web server which can also be used as a reverse proxy, load balancer, mail proxy and HTTP cache.

> The goal behind NGINX was to create a very fast web server for handling many concurrent connections.

Reviewer: ```suggestion
The goal behind NGINX was to create a fast web server for handling a large amount of concurrent connections.
```

NGINX has a few common use cases:

> - NGINX can serve static files (such as HTML, CSS and JavaScript)

Reviewer: ```suggestion
- Static files: NGINX can serve static files, such as HTML, CSS and JavaScript.
```

- Act as a reverse proxy: handle and process all requests before sending them to another application server.
- Be used as a load balancer: requests can be distrubed over multiple instances of your application to allow for high availability should one of you application servers fail.
## Installation
### Linux
On Linux systems, NGINX can usually be installed using the relevant package manager:
#### CentOS/RHEL
```bash
sudo yum install -y nginx
```
#### Debian/Ubuntu
```bash
sudo apt install -y nginx
```
### Windows
An installer for NGINX can be downloaded from the [NGINX downloads page](http://nginx.org/en/download.html).
- Download the latest, stable version of NGINX
- Unzip the downloaded file
- Run the `nginx.exe` executable - don't worry if it seems like nothing happens, NGINX will run in the background.

### Docker
If you have Docker configured then you can get NGINX running using the `nginx` image:
```bash
docker run -d --rm -p 80:80 --name nginx nginx
```

> Once you have installed NGINX for your relevant Operating System, you can then navigate to `http://localhost` in your web browser to see if NGINX is running, if you are on Linux and don't have a GUI, then you can run `curl http://localhost` to see if you can get a response from the NGINX service.

Reviewer: ```suggestion
Once you have installed NGINX for your relevant Operating System, you can navigate to `http://localhost` in your web browser to see if NGINX is running. If you are on Linux and don't have a GUI, then you can run `curl http://localhost` to see if you can get a response from the NGINX service.
```
